pkg/server/metrics/metrics.go

```diff
@@ -284,6 +299,11 @@ func (s *ServerMetrics) SetBackendCount(count int) {
 	s.backend.WithLabelValues().Set(float64(count))
 }
 
+// SetTotalBackendCount sets the total number of backend connection.
+func (s *ServerMetrics) SetTotalBackendCount(managerName string, count int) {
+	s.totalBackendCount.WithLabelValues(managerName).Set(float64(count))
```

Comment:
Labels can generate cardinality issues. I don't think this is likely to be a problem but we can reassure/protect ourselves by changing managerName from a string (unbounded cardinality) to an enumeration with an observation on limiting the number of options.

Reply:
Done.

Based on @jkh52 's suggestion above, I used proxy strategy enum as the label. (Thank you @jkh52 !)
It introduced the circular dependency, so I extracted out ProxyStrategy related types and functions into a separate package.